src/table/system.rs

```diff
@@ -229,3 +237,34 @@ struct SystemTableImpl {
 impl<View: SystemTableView> super::Table for SystemTable<View> {
     const SIGNATURE: u64 = 0x5453_5953_2049_4249;
 }
+
+impl Debug for SystemTableImpl {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        // it's not that easy to get the C16String to an actual printable string
+        // Therefore we format it separately
+        let mut vendor_buf = arrayvec::ArrayString::<256>::new();
+        write!(vendor_buf, "{}", unsafe {
+            CStr16::from_ptr(self.fw_vendor)
+        })
+        .unwrap();
```

Comment:
Why not add a debug impl to `C16String`?

Reply:
I did. Problem now is:
```
[DEBUG]:   src/lib.rs@114: UefiSystemTable {
[DEBUG]:   src/lib.rs@114:     header: Header {
[DEBUG]:   src/lib.rs@114:         signature: 0x0000000000000078,
[DEBUG]:   src/lib.rs@114:         revision: 2.7.0,
[DEBUG]:   src/lib.rs@114:         size: 120,
[DEBUG]:   src/lib.rs@114:         crc: 3701267086,
[DEBUG]:   src/lib.rs@114:     },
[DEBUG]:   src/lib.rs@114:     fw_vendor: CStr16(['E', 'D', 'K', ' ', 'I', 'I', '\u{0}']),
[DEBUG]:   src/lib.rs@114:     fw_revision: 1.0.0,
```

It's not a nice string for the `fw_vendor`-field and I don't know a nice approach to get a display-formatting there without ugly code.

Reply:
Well, considering a debug display isn't necessarily meant to be "pretty", I guess it's ok?

The alternative would be to come up with a `Display` impl for `CStr16` and use that instead.

Reply:
Yeah, it's okay. I didn't find a possible and nice way there to use the display-formatting of the specific field when I use  `debug_struct` for all the other fields. I think it's okay now.